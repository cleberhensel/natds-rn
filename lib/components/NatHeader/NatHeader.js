import React from 'react';
import {
  Image, Platform, Text, TouchableOpacity,
} from 'react-native';
import PropTypes from 'prop-types';

import {
  HeaderWrapper,
  HeaderTitle,
  HeaderIcon,
  HeaderRight,
  AndroidStyle,
} from './NatHeader.styles';
import { icons } from '../../assets/theme/icons';

export const NatHeader = ({
  title, icon, onPress, textRight, iconRight, onPressRight,
}) => {
  if (!title) {
    throw new Error('Title should not empty string');
  }

  return (
    <HeaderWrapper
      style={Platform.OS === 'android' && AndroidStyle.android}
      accessible
      accessibilityRole="header"
    >
      {icon && (
      <HeaderIcon
        onPress={onPress}
        accessibilityRole="button"
      >
        <Image
          source={icons[icon]}
          accessible
          accessibilityLabel="navegação para retornar a página anterior"
        />
      </HeaderIcon>
      )}

      <HeaderTitle
        icon={icon}
        accessibilityLabel={`${title}`}
      >
        {title}
      </HeaderTitle>
      <HeaderRight accessibilityRole="button">
        <TouchableOpacity onPress={onPressRight}>
          <Text style={{ color: 'orange', fontSize: 14 }}>
            {' '}
            {textRight}
            {' '}
          </Text>
          <Image
            accessible
            source={icons[iconRight]}
            accessibilityLabel="realizar ação do item da barra superior direita"
          />
        </TouchableOpacity>
      </HeaderRight>
    </HeaderWrapper>
  );
};

NatHeader.propTypes = {
  icon: PropTypes.string,
  iconRight: PropTypes.string,
  onPress: PropTypes.func.isRequired,
  onPressRight: PropTypes.func,
  textRight: PropTypes.string,
  title: PropTypes.string.isRequired,
};

NatHeader.defaultProps = {
  icon: '',
};
