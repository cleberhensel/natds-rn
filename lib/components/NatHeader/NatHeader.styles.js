import { Platform, StyleSheet } from 'react-native';
import styled, { css } from 'styled-components/native';
import { colors } from '../../assets/theme/colors/colors';
import { natura } from '../../assets/theme/fonts/fonts';

export const AndroidStyle = StyleSheet.create({
  android: {
    marginTop: 4,
  },
});

export const HeaderWrapper = styled.View`
  flex-wrap: wrap;
  width: 100%;
  height: 56;
  padding-top: 16;
  padding-bottom: 16;
  padding-left: 16;
  background-color: ${colors.primary.white};
  ${Platform.select({
    ios: css`
      flex-direction: row;
      justify-content: space-between;
    `,
    android: css`
      flex-direction: row;
    `,
  })}
`;

export const HeaderIcon = styled.TouchableOpacity`
  width: 24;
  ${Platform.select({
    ios: css`
      flex: 1;
      justify-content: flex-start;
    `,
  })}
`;

export const HeaderTitle = styled.Text`
  margin-left: 24;
  font-size: 20;
  color: ${colors.bw.grayDarker};
  font-family: ${natura.regular};
`;

export const HeaderRight = styled.View`
  width: 60;
  color:${colors.primary.yellow};
  ${Platform.select({
    ios: css`
      flex: 1;
      justify-content: flex-end;
    `,
    android: css`
      flex:1;
      align-items: flex-end;
      margin-right:14;
    `,
  })
}
`;
