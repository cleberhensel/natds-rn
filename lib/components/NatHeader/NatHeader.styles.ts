import { Platform, StyleSheet } from "react-native";
import styled, { css } from "styled-components/native";
import { colors } from "../../assets/theme/colors/colors";
import { natura } from "../../assets/theme/fonts/fonts";
export const AndroidStyle = StyleSheet.create({
  android: {
    marginTop: 4
  }
});
export const HeaderWrapper = styled.View`
  display: flex;
  flex-wrap: nowrap;
  flex-direction: row;
  justify-content: space-between;
  align-items: center;
  width: 100%;
  height: 56;
  padding-top: 0;
  padding-bottom: 0;
  padding-left: 16;
  padding-right: 16;
  background-color: ${colors.primary.white};
  ${Platform.select({
    android: css`
      flex-direction: row;
    `,
    ios: css`
      flex-direction: row;
      justify-content: space-between;
    `
  })}
`;
export const HeaderIcon = styled.TouchableOpacity`
  width: 24;
`;
export const HeaderTitle = styled.Text`
  margin-left: 24;
  font-size: 20;
  color: ${colors.bw.grayDarker};
  font-family: ${natura.regular};
  ${Platform.select({
    android: css`
      padding-left: 16px;
    `,
    ios: css`
      padding-left: 45px;
      padding-top: 8px;
    `
  })}
`;
export const HeaderRight = styled.View`
  width: 60;
  color: ${colors.primary.yellow};
  ${Platform.select({
    android: css`
      flex: 1;
      align-items: flex-end;
      margin-right: 8;
    `,
    ios: css`
      margin-top: 4;
    `
  })}
`;
