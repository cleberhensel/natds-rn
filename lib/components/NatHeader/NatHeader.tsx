import React from "react";
import { Image, Platform, Text, TouchableOpacity } from "react-native";
import {
  HeaderWrapper,
  HeaderTitle,
  HeaderIcon,
  HeaderRight,
  AndroidStyle
} from "./NatHeader.styles";
import { icons } from "../../assets/theme/icons";
type NatHeaderProps = {
  icon?: string,
  iconRight?: string,
  onPress: (...args: any[]) => any,
  onPressRight?: (...args: any[]) => any,
  textRight?: string,
  title: string
};
export const NatHeader: React.SFC<NatHeaderProps> = ({
  title,
  icon,
  onPress,
  textRight,
  iconRight,
  onPressRight
}) => {
  if (!title) {
    throw new Error("Title should not empty string");
  }
  return (
    <HeaderWrapper
      style={Platform.OS === "android" && AndroidStyle.android}
      accessibilityRole="header"
    >
      {icon && (
        <HeaderIcon onPress={onPress} accessibilityRole="button">
          <Image
            source={icons[icon]}
            accessibilityLabel="navegação para retornar a página anterior"
          />
        </HeaderIcon>
      )}

      <HeaderTitle icon={icon} accessibilityLabel={`${title}`}>
        {title}
      </HeaderTitle>
      <HeaderRight accessibilityRole="button">
        <TouchableOpacity onPress={onPressRight}>
          <Text
            accessibilityLabel={textRight}
            style={{ color: "orange", fontSize: 14 }}
          >
            {" "}
            {textRight}{" "}
          </Text>
          <Image
            source={icons[iconRight]}
            accessibilityLabel="realizar ação do item da barra superior direita"
          />
        </TouchableOpacity>
      </HeaderRight>
    </HeaderWrapper>
  );
};
NatHeader.defaultProps = {
  icon: ""
};
