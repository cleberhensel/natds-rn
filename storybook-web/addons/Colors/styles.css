.colors {
  font-weight: 400;
  line-height: 1.5;
}

.colors__container {
  max-width: 1400px;
  display: flex;
  flex-flow: row wrap;
  padding: 60px;
}

.colors__group {
  display: flex;
  flex-flow: row wrap;
  flex: 0 0 100%;
  margin-bottom: 30px;
}

.colors__group__title {
  font-size: 1.5rem;
  font-family: Roboto, Arial, Helvetica, sans-serif;
  font-weight: 400;
  color: #555;
  flex: 0 0 100%;
  padding: 0 0 20px;
  text-transform: capitalize;
  margin: 0;
}

.colors__group__list {
  flex: 0 0 100%;
  display: flex;
  flex-flow: row wrap;
  padding: 0;
  margin: 0;
}

.colors__item {
  width: 130px;
  display: flex;
  flex: 0 0 auto;
  position: relative;
  flex-flow: column;
  overflow: hidden;
  padding: 0;
  margin: 0 15px 15px 0;
}

.colors__item__color {
  width: 130px;
  height: 130px;
  margin-bottom: 5px;
}

.colors__item__color-name,
.colors__item__name {
  font-family: Roboto, Arial, Helvetica, sans-serif;
  font-size: 0.875rem;
  line-height: 1rem;
  padding: 0;
  margin: 0;
}

.colors__item__name {
  color: #888;
  margin-bottom: 5px;
}

.colors__item__color-name {
  color: #555;
}
