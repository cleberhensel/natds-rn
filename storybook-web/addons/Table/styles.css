.table__container {
  font-weight: 400;
  line-height: 1.5;
  max-width: 800px;
  width: 100%;
  font-family: Roboto, Arial, Helvetica, sans-serif;
  color: #555;
  margin: 60px;
}

.table__title {
  font-size: 18px;
  font-weight: 400;
  text-transform: capitalize;
  padding: 0 0 20px;
  margin: 0;
}

.table {
  width: 100%;
  border-collapse: collapse;
}

.table__body__row {
  background: #f9f9f9;
  border-bottom: #fff 1px solid;
  padding: 10px;
}

.table__head__item:first-child {
  width: 25%;
}

.table__head__item,
.table__body__item {
  font-size: 0.875rem;
  text-align: left;
}

.table__head__item {
  padding-bottom: 10px;
  text-transform: capitalize;
}

.table__body__item {
  line-height: 1.25rem;
  font-weight: 400;
  height: 20px;
  padding: 10px;
}

.table__body__item__color {
  border: #ddd 2px solid;
  border-radius: 50%;
  width: 16px;
  height: 16px;
  display: inline-block;
  vertical-align: top;
  margin: 2px 10px 0;
}

.table__body__item__color-name {
  line-height: 1rem;
  display: inline-block;
}
