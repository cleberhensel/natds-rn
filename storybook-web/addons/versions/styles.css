.versions {
  font-weight: 400;
  line-height: 1.5;
  font-family: Roboto, Arial, Helvetica, sans-serif;
  color: #555;
  margin: 30px;
}

.versions__type {
  padding: 0 0 10px;
  margin: 0 10px 10px;
}

.versions__type__name {
  border-bottom: rgba(0,0,0,.1) 1px solid;
  font-size: 18px;
  text-transform: capitalize;
  padding: 0 0 10px;
  margin: 0 0 10px;
}

.versions__type__warning {
  font-size: 18px;
  text-transform: capitalize;
  margin-bottom: 10px;
}

.versions__type__button {
  background: rgba(0,0,0,.3);
  border: 0;
  border-radius: 4px;
  font-size: 1rem;
  color: #fff;
  text-transform: capitalize;
  cursor: pointer;
  padding: 7px 14px;
  margin: 0 10px 10px 0;
}

.versions__type__button:hover {
  background: rgba(0,0,0,.2);
}

.versions__type__button:active {
  background: rgba(0,0,0,.25);
}

.versions__type__button--selected,
.versions__type__button--selected:hover {
  background: rgba(0,0,0,.6);
  cursor: default;
}
