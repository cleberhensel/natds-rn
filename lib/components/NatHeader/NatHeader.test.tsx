import React from "react";
import renderer from "react-test-renderer";
import { shallow } from "enzyme";
import { NatHeader } from "./NatHeader";
import { HeaderTitle } from "./NatHeader.styles";
describe("NatHeader component", () => {
  it("Should render container component correctly", () => {
    const header = renderer
      .create(<NatHeader title="Title test" onPress={jest.fn()} />)
      .toJSON();
    expect(header).toMatchSnapshot("NatHeader snapshot");
  });
  it("Should render header component with title", () => {
    const wrapper = shallow(
      <NatHeader title="Title test" onPress={jest.fn()} />
    );
    expect(wrapper.find(HeaderTitle).text()).toBe("Title test");
  });
  it("Should throw exception when header component title is empty", () => {
    expect(() => shallow(<NatHeader title="" onPress={jest.fn()} />)).toThrow(
      "Title should not empty string"
    );
  });
});
