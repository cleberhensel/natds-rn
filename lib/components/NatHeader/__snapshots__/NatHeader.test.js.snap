// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NatHeader component Should render container component correctly: NatHeader snapshot 1`] = `
<View
  accessibilityRole="header"
  style={
    Array [
      Object {
        "alignItems": "center",
        "backgroundColor": "#FFFFFF",
        "display": "flex",
        "flexDirection": "row",
        "flexWrap": "nowrap",
        "height": 56,
        "justifyContent": "space-between",
        "paddingBottom": 0,
        "paddingLeft": 16,
        "paddingRight": 16,
        "paddingTop": 0,
        "width": "100%",
      },
      false,
    ]
  }
>
  
  <Text
    accessibilityLabel="Title test"
    icon=""
    style={
      Array [
        Object {
          "color": "#333333",
          "fontFamily": "NaturaSans-Regular",
          "fontSize": 20,
          "marginLeft": 24,
          "paddingLeft": 45,
          "paddingTop": 8,
        },
      ]
    }
  >
    Title test
  </Text>
  <View
    accessibilityRole="button"
    style={
      Array [
        Object {
          "color": "#F4AB34",
          "marginTop": 4,
          "width": 60,
        },
      ]
    }
  >
    <View
      accessible={true}
      focusable={false}
      isTVSelectable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "orange",
            "fontSize": 14,
          }
        }
      >
         
         
      </Text>
      <Image
        accessibilityLabel="realizar ação do item da barra superior direita"
      />
    </View>
  </View>
</View>
`;
